crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -705,6 +736,9 @@ pub trait SemanticSyntaxContext {
     /// The target Python version for detecting backwards-incompatible syntax changes.
     fn python_version(&self) -> PythonVersion;
 
+    /// Return the [`TextRange`] at which a name is declared as `global`.
+    fn global(&self, name: &str) -> Option<TextRange>;
```

Comment:
Red Knot doesn't support globals yet, so it's hard to say if this will be supported in Red Knot and how ;)

It does have the downside that this check will not work in the test context. And it also means that it requires one extra traversal of the body to get the globals first but I don't see how we can avoid that.

But we (you) can worry about that later ;)